### DISCUSSION TITLE
What is the best way to send struct data from a rust backend to a ts frontend with types generated from this crate?

### DISCUSSION THREAD

**Discussion by MrVintage710** [Aleph-Alpha/ts-rs#348] [Q&A] [answered]
I know that there are many ways, but I am wondering if there is a way that is supported by this library. I am considering using this crate for an open source project I am starting.

**Comment by NyxCode** [comment, marked as answer]
The most common use-case is for REST APIs sending JSON.

ts-rs is agnostic about this, though - it just matches the serde data model, so you can chose a different format if you wanted.  
You could, of course, also use WebSockets and send JSON (or anything else) over them as well.

Again, `ts-rs` just gives you TS bindings - that's it.

**Reply by MrVintage710** [reply]
Awesome, thanks for the answer! I think that we are going to use it in a Tauri app that we are writing, and Tauri already uses `serde-json` in there messaging pipeline, so this is perfect! I'll let you know how it goes.
